src/run_program.rs

```diff
@@ -327,27 +323,27 @@ impl<'a, D: Dialect> RunProgramContext<'a, D> {
         self.eval_pair(program, env)
     }
 
-    fn parse_softfork_arguments(
-        &self,
-        args: NodePtr,
-    ) -> Result<(OperatorSet, NodePtr, NodePtr), EvalErr> {
+    fn parse_softfork_arguments(&self, args: NodePtr) -> Result<(OperatorSet, NodePtr, NodePtr)> {
         let [_cost, extension, program, env] = get_args::<4>(self.allocator, args, "softfork")?;
 
         let extension =
             self.dialect
                 .softfork_extension(uint_atom::<4>(self.allocator, extension, "softfork")? as u32);
         if extension == OperatorSet::Default {
-            err(args, "unknown softfork extension")
+            Err(EvalErr::UnknownSoftforkExtension)
         } else {
             Ok((extension, program, env))
         }
     }
 
-    fn apply_op(&mut self, current_cost: Cost, max_cost: Cost) -> Result<Cost, EvalErr> {
+    fn apply_op(&mut self, current_cost: Cost, max_cost: Cost) -> Result<Cost> {
         let operand_list = self.pop()?;
         let operator = self.pop()?;
         if self.env_stack.pop().is_none() {
-            return err(operator, "runtime error: env stack empty");
+            return Err(EvalErr::InternalError(
+                operator,
+                "Environment Stack Empty".to_string(),
+            ))?;
```

Comment:
```suggestion
            ));
```

Does this `?` do anything?

Reply:
nope, good catch.